core/providers/java/jdk.go

```diff
@@ -0,0 +1,28 @@
+package java
+
+import (
+	"strconv"
+
+	"github.com/railwayapp/railpack/core/generate"
+)
+
+const DEFAULT_JDK_VERSION = "21"
+
+func (p *JavaProvider) setJDKVersion(ctx *generate.GenerateContext, miseOut *generate.MiseStepBuilder) error {
+	miseStep := ctx.GetMiseStepBuilder()
+	if miseOut == nil {
+		miseOut = miseStep
+	}
```

Comment:
I am a little confused about this. Why is `miseOut` needed? Can you just use the `miseStep` directly

Reply:
`miseOut` gets the runtime dependencies `MiseStepBuilder` for runtime - since I don't need Gradle or Maven for the runtime, just Java, it makes sense (to me) to have a separate mise step for that.

Reply:
This makes sense. I mentioned it in another comment, but can you always pass in the miseStep (either for build or runtime deps). Having it optional and then defaulting to the build deps is a bit confusing imo